### DISCUSSION TITLE
Display some questions only on mobile apps

### DISCUSSION THREAD

**Discussion by raphael0202** [openfoodfacts/robotoff#1243] [Ideas]
### Problem
Currently, we don't discriminate against clients (Hunger Games/mobile apps) concerning questions: all clients get the same questions if they request the same question type(s).

However, some questions require to have a physical instance of the product. If an insight is generated from an image and the image is more than 120 days older than the most recent image, we mark the insight as not automatically processable.
Therefore, we have many insights in which we're confident (they're usually automatically applied), but which we display in Hunger Games because we're not sure that the information didn't change (as there are newer pictures available).
We should restrict these questions to clients for whom we're certain they have a physical instance of the product (i.e. mobile apps).

The current status is not satisfactory, as it provides no added value to display these questions to users on Hunger-Games/Product Opener (and it wastes contributor time).

### Proposed solution

Add a new `distribution_channels` field in `ProductInsight.data` that contains the channels we can broadcast the questions to:
- `mobile`
- `web` (Product Opener)
- `hunger-games`

Add in all question endpoints a new `channel` parameter to select the channel. As a default (if no `channel` is provided), we don't filter on distribution channels.

**Comment by alexgarel** [comment]
@raphael0202 in the same spriti, I think it would also be worth storing from where a question was answered. But it means apps should send who they are along with the answer.
